revpi_flat.c

```diff
@@ -346,6 +352,19 @@ int revpi_flat_init(void)
 		return -ENXIO;
 	}
 
+	flat->button_desc = gpio_to_desc(REVPI_FLAT_BUTTON_GPIO);
```

Comment:
If I understand this right, integer based GPIOs are deprecated. As this method is used in the other parts of the code as well this should be ok for now. But for future development we should use the new interfaces.

Reply:
Since we do not grab the gpio exclusiveley the gpio-key driver (or any other interested instance) can also make use of it.